src/backend/opencl/clfft.cpp

```diff
@@ -156,7 +160,7 @@ SharedPlan findPlan(clfftLayout iLayout, clfftLayout oLayout, clfftDim rank,
     // CommandQueue() returns the actual cl_command_queue handle
     CLFFT_CHECK(clfftBakePlan(*temp, 1, &(opencl::getQueue()()), NULL, NULL));
 
-    retVal.reset(temp, [](PlanType *p) {
+    retVal.reset(temp.release(), [](PlanType *p) {
```

Comment:
Making a unique_ptr and then release, feels like it is unnecessary here.

Reply:
if any of those CLFFT_CHECK fails it will leak otherwise. This is the exception safe way of doing it.